mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -84,7 +87,20 @@ abstract class StreamingLinearAlgorithm[
     }
     data.foreachRDD { (rdd, time) =>
       if (!rdd.isEmpty) {
-        model = Some(algorithm.run(rdd, model.get.weights))
+        val newModel = algorithm.run(rdd, model.get.weights)
+
+        val numNewDataPoints = rdd.count()
+        val discount = getDiscount(numNewDataPoints)
+
+        val updatedDataWeight = previousDataWeight * discount + numNewDataPoints
+        val lambda = numNewDataPoints / math.max(updatedDataWeight, 1e-16)
```

Comment:
Assuming `previousDataWeight >= 0` and `discount >= 0`, `updatedDataWeight >= numNewDataPoints` so there's no need for protecting against div by zero here.

We should remove the check but keep a comment here in case readers of the code worry about it